src/modules/block/index.ts

```diff
@@ -0,0 +1,308 @@
+    }
+
+    const updatedProp = Object.keys(newProps).find(
+      (prop) =>
+        (oldProps as Record<string, any>)[prop] === undefined ||
+        (oldProps as Record<string, any>)[prop] !==
+          (newProps as Record<string, any>)[prop],
+    );
+
+    return !!updatedProp;
+  }
+
+  setProps = (nextProps: Partial<P>) => {
+    if (!nextProps) {
+      return;
+    }
+
+    Object.assign(this.props, nextProps);
+  };
+
+  get element() {
+    return this._element;
+  }
+
+  _addEvents() {
+    const { events = {} } = this.props as unknown as Record<string, any>;
+
+    if (!this._element) {
+      return;
+    }
+
+    Object.keys(events).forEach((eventName) => {
+      this._element!.addEventListener(eventName, events[eventName]);
+    });
+  }
+
+  _removeEvents() {
+    const { events = {} } = this.props as unknown as Record<string, any>;
+
+    if (!this._element) {
+      return;
+    }
+
+    Object.keys(events).forEach((eventName) => {
+      this._element!.removeEventListener(eventName, events[eventName]);
+    });
+  }
+
+  _render() {
+    const element = this.render();
+
+    this._removeEvents();
+
+    if (!this._element) {
+      this._element = element;
+    } else {
+      this._element.replaceWith(element);
+
+      this._element = element;
+    }
+
+    this._addEvents();
+  }
+
+  render(): HTMLElement {
+    return new HTMLElement();
+  }
+
+  compile(props: P): HTMLElement {
+    const propsAndStubs = { ...props };
+
+    Object.entries(this.children).forEach(([key, child]) => {
+      if (Array.isArray(child)) {
+        (propsAndStubs as Record<string, any>)[key] = child.map((subChild) => {
+          if (this._isBlocksObject(subChild)) {
+            return Object.keys(subChild).reduce((acc, k) => {
+              acc[k] = `<div data-id="${
+                (subChild as BlocksObject)[k]._id
+              }"></div>`;
+              return acc;
+            }, {} as Record<string, string>);
+          }
+          return `<div data-id="${(subChild as Block)._id}"></div>`;
+        });
+      } else {
+        (propsAndStubs as Record<string, any>)[
+          key
+        ] = `<div data-id="${child._id}"></div>`;
+      }
+    });
+
+    const fragment = document.createElement('template');
+    const render = Handlebars.compile(this._template, { noEscape: true });
+    fragment.insertAdjacentHTML('afterbegin', render(propsAndStubs));
+
+    const element = fragment.firstElementChild as HTMLElement;
+
+    Object.values(this.children).forEach((child) => {
+      if (Array.isArray(child)) {
+        child.forEach((subChild) => {
+          if (this._isBlocksObject(subChild)) {
+            Object.keys(subChild).forEach((k) => {
+              const block = (subChild as BlocksObject)[k];
+              const stub = element.querySelector(`[data-id="${block._id}"]`);
+
+              if (!stub || !block.element) {
+                return;
+              }
+
+              stub!.replaceWith(block.element);

```

Comment:
Из-за условия ТС не должен ругаться и `!` можно опустить
```suggestion
              stub.replaceWith(block.element);
```